### DISCUSSION TITLE
Unable to compile or run local copy of this library, too many compilation errors?

### DISCUSSION THREAD

**Discussion by aloksharma1** [SixLabors/ImageSharp.Web#300] [Q&A] [answered]
Hello,
i wanted to switch from Skiasharp to Imagesharp, my problem is that i have and existing project with self written middleware that does resizing for my project using Skiasharp lib and thus we are using our own query format to grab resized images.
now i wanted to switch to imagesharp due to performance, stability and other factors but i got stuck on the part where you call images query by a different name like `http://myexampledomain.com/image1.jpg?width=xxx` while we call it with different query params i.e. `w for width`.
so i though i can simple override or add more query params to your code, but i am getting way too many errors for such simple stuff.

please help

**Comment by tocsoft** [comment]
@aloksharma1 you haven't actually provided any useful information in your initial comment.... please provide some details like, how your trying to compile the code, what versions of dotnet you are running it against and the most critical thing you should have provided what specific errors you are actually seeing (Line numbers, error codes etc i.e. details from the build output). 

As you can see from our last build (as of writing) of the `main` branch  the library builds with zero warnings/errors https://github.com/SixLabors/ImageSharp.Web/actions/runs/3740881127

**Comment by JimBobSquarePants** [comment, marked as answer]
@aloksharma1   It sounds like you are not installing the git submodule which contains the validation rules and other helper classes. 

I've copied the git instructions from the readme to allow that. 

```
git submodule update --init --recursive
```

However. It sounds to me like you simply want to map commands. You can do that in `ImageSharpMiddlewareOptions.OnParseCommandsAsync` event at startup. 

On this documentation page you can see an example of how to manipulate the commands recognized by the middleware.

https://docs.sixlabors.com/articles/imagesharp.web/gettingstarted.html#setup-and-configuration

**Comment by aloksharma1** [comment]
thanks for the reply, it seems apart from shared module there was something wrong with my local copy after deleting it and cloning again, my copy started compiling normally.  Another thing i had to fire `git config --system core.longpaths true` this again.

i am checking the docs, will come back again if i get stuck somewhere.
